lib/devices/ios/remote-debugger.js

```diff
@@ -449,8 +459,21 @@ RemoteDebugger.prototype.waitForDom = function (cb) {
 
 RemoteDebugger.prototype.waitForFrameNavigated = function (cb) {
   this.logger.debug("Waiting for frame navigated...");
+  var startMs = Date.now();
+
+  // wrapping cb to be able to log some data
+  var _cb = cb;
+  cb = function () {
+    this.logger.debug('frame navigated in ' + ((Date.now() - startMs)/1000) + ' sec.' );
+    var args = [].slice.call(arguments);
+    _cb.apply(null, args);
+  }.bind(this);
+
   this.frameNavigatedCbs.push(cb);
-  this.navigatingTimeout = setTimeout(this.frameNavigated.bind(this), 500);
+  this.navigatingTimeout = setTimeout(function () {
+     this.logger.debug('frame navigated timeout triggered');
+     this.frameNavigated();
+  }.bind(this), this.useConservativePageLoadTimeouts ? 30000 : 500);
```

Comment:
No, the delay is the other timeout (line 371), this is the max time we wait. 

We should probably properly implement pageLoadTimeout properly, to be consistent with Selenium.